rdkPlugins/Storage/source/Storage.cpp

```diff
@@ -503,6 +493,129 @@ std::vector<DynamicMountProperties> Storage::getDynamicMounts() const
+ *
+ *
+ *  @return vector of MountOwnerDetails that were in the config
+ */
+std::vector<std::unique_ptr<MountOwnerDetails>> Storage::getMountOwnerDetails() const
+{
+    AI_LOG_FN_ENTRY();
+
+    const std::vector<MountOwnerProperties> mountOwners = getMountOwners();
+    std::vector<std::unique_ptr<MountOwnerDetails>> ownerDetails;
+
+    // Setup the user/group IDs
+    uid_t uid = 0;
+    gid_t gid = 0;
+    setupOwnerIds(uid, gid);
+
+    // loop though all the mount owners for the given container and create individual
+    // MountOwnerDetails objects for each
+    for (const MountOwnerProperties &properties : mountOwners)
+    {
+        // create the mount owner details and make sure it was constructed
+        auto mountOwner = std::make_unique<MountOwnerDetails>(mRootfsPath,
+                                                              properties,
+                                                              uid,
+                                                              gid,
+                                                              mUtils);
+
+        if (mountOwner)
+        {
+            ownerDetails.emplace_back(std::move(mountOwner));
+        }
+    }
+
+    AI_LOG_FN_EXIT();
+
+    return ownerDetails;
+}
+
+// -----------------------------------------------------------------------------
+/**
+ *  @brief Reads container config and creates all dynamic mounts in DynamicMountProperties
+ *  type objects.
+ *
+ *
+ *  @return vector of DynamicMountProperties that were in the config
+ */
+std::vector<MountOwnerProperties> Storage::getMountOwners() const
+{
+    AI_LOG_FN_ENTRY();
+
+    std::vector<MountOwnerProperties> mountOwners;
+
+    // Check if container has mount data
+    if (mContainerConfig->rdk_plugins->storage->data)
+    {
+        // loop though all the mount owners for the given container and create individual
+        // MountOwnerProperties objects for each
+        for (size_t i = 0; i < mContainerConfig->rdk_plugins->storage->data->mount_owner_len; i++)
+        {
+            auto mountOwner = mContainerConfig->rdk_plugins->storage->data->mount_owner[i];
+
+            MountOwnerProperties mountOwnerProps;
+            mountOwnerProps.source = std::string(mountOwner->source);
+            if (mountOwner->user)
+            {
+                mountOwnerProps.user = std::string(mountOwner->user);
+            }
+            if (mountOwner->group)
+            {
+                mountOwnerProps.group  = std::string(mountOwner->group);
+            }
+            mountOwnerProps.recursive = mountOwner->recursive_present ? mountOwner->recursive : false;
+            mountOwners.push_back(mountOwnerProps);
+        }
+    }
+    else
+    {
+        AI_LOG_ERROR("No storage data in config file");
+    }
+
+    AI_LOG_FN_EXIT();
+    return mountOwners;
+}
+
+// -----------------------------------------------------------------------------
+/**
+ *  @brief Gets userId and groupId
```

Comment:
Can you change the description into more verbose one i.e. `Gets userId and groupId of container owner and maps it into host user namespace.`

Reply:
Changed commet to be more accurate but this function does not perform any mapping itself or changing of ownership itself.